### PR TITLE
separate log formatter and log location options

### PR DESCRIPTION
**Pull request by grosser**
this file was confusing and lograge and syslog don't need to be the same ... so now I can debug lograge locally without having to use syslog

### CODE REVIEW

File: lib/samson/logging.rb [hunk 1]
```diff
@@ -1,33 +1,19 @@
...
-
+  # good for production hosts with syslog
+  # TODO: add Syslog::Logger#silence to support dev mode
+  # TODO: add tagged logging so we get request-ids
   config.logger = Syslog::Logger.new('samson')
-  config.action_cable.logger = Syslog::Logger.new('samson')
-  config.lograge.formatter = Lograge::Formatters::Logstash.new
-  config.logger.formatter = config.action_cable.logger.formatter = Samson::SyslogFormatter.new
-elsif ENV["SERVER_MODE"] # regular file logger that needs rotating
+  config.logger.formatter = Samson::SyslogFormatter.new
```
Comment (sathishavm) [inline lib/samson/logging.rb]: Need to add formatter for 'action_cable',  Samson strictly needs to log in json format if it is RAILS_LOG_TO_SYSLOG, But `action_cable` isn't doing it.

File: lib/samson/logging.rb [hunk 2]
```diff
@@ -46,7 +32,25 @@
   end
 end
 
-# workaround for our server ... always gets SIGHUP even though it does not do file logging
-if ENV["IGNORE_SIGHUP"]
-  Signal.trap(:SIGHUP) {}
+# log 1 message per request to in json/syslog format
+if ENV["RAILS_LOG_WITH_LOGRAGE"]
```
Comment (sathishavm) [inline lib/samson/logging.rb]: Need to add in .env.example with explanation

File: lib/samson/logging.rb (hunk 1 above)
Comment (sathishavm) [reply lib/samson/logging.rb]: we may break elk!!

Comment (grosser) [reply lib/samson/logging.rb]: verified in staging that it still does
```
{"level":1,"@timestamp":"2019-03-27T22:17:38.152+00:00","application":"samson","host":"XXX","message":"Successfully upgraded to WebSocket (REQUEST_METHOD: GET, HTTP_CONNECTION: upgrade, HTTP_UPGRADE: websocket)"}
```

File: lib/samson/logging.rb (hunk 2 above)
Comment (grosser) [reply lib/samson/logging.rb]: added

**Comment by sathishavm** [comment]
Pls see comments in https://zendesk.atlassian.net/browse/BRE-218, Some action cable logs is not in json format, that's why I added it!!

**Comment by grosser** [comment]
https://github.com/zendesk/samson/pull/2871 added the actioncabble logger but did not include repro instructions ... all I can see is that syslog actioncable things are coming in on staging which uses syslog and locally if I `bundle open` syslog and `puts` in `syslog/logger.rb` instead of writing to syslog

**Pull request merged**